deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/conf/NeuralNetConfiguration.java

```diff
@@ -1245,8 +1246,14 @@ public Builder convolutionMode(ConvolutionMode convolutionMode) {
          * @param constraints Constraints to apply to all layers
          */
         public Builder constraints(LayerConstraint... constraints){
-            return constraints(Arrays.asList(constraints));
-        }
+            List<LayerConstraint> currentConstraints = this.constraints;
+            for(LayerConstraint c : constraints ){
+                BaseConstraint constraint = (BaseConstraint) c.clone();
+                constraint.setApplyToBiases(true);
+                constraint.setApplyToWeights(true);
+                currentConstraints.add(constraint);
+            }
+            return constraints(currentConstraints);         }
```

Comment:
I'm not sure defaulting  to biases on the .constraints(...) method is a good idea?
I'd probably go for .constraints(LayerContstraint...) method requiring the user to manually specify the weight/bias keys (i.e., the weightConstraints and biasConstraint methods become the 'default' ones to use).

This current approach will also override whatever the user has set...

Reply:
@AlexDBlack what do you mean by "what the user has set"? This IS what the user sets. You mean what he sets on individual layers? There has to be some rule for what takes precedence and I think it makes sense that global config overrides local.